session_0/item.rb

```diff
@@ -51,8 +47,8 @@ def current_price
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
-    stock_price = Hash.new
-    (1..4).each {|x| stock_price[x] = 0}
+    stock_price = {}
+    1.upto(Item::CATEGORIES.count) {|x| stock_price[x] = 0}
```

Comment:
What if some categories were discounted, something like:
```ruby
{
  1 => 'Books',
  2 => 'Electronics',
  4 => 'Hobbies',
  42 => 'Clothes'
}
```

Reply:
Yes, but the `.count` will only count the number of elements in `CATEGORIES`, so it doesn't matter what the key values are.

Reply:
1.upto(4) would iterate and create the stock_price as follows:

```ruby
{1 => 0, 2 => 0, 3 => 0, 4 => 0}
```

What we want is:
```ruby
{1 => 0, 2 => 0, 4=> 0, 42 => 0}
```